eng/pipelines/common/templates/runtimes/build-test-job.yml

```diff
@@ -51,12 +52,18 @@ jobs:
       continueOnError: true
 
     ${{ if eq(parameters.testGroup, 'innerloop') }}:
-      name: '${{ parameters.runtimeFlavor }}_common_test_build_p0_${{ parameters.osGroup }}${{ parameters.osSubgroup }}_${{ parameters.archType }}_${{ parameters.buildConfig }}'      
-      displayName: '${{ parameters.runtimeFlavorDisplayName }} Common Pri0 Test Build ${{ parameters.osGroup }}${{ parameters.osSubgroup }} ${{ parameters.archType }} ${{ parameters.buildConfig }}'
-
+      testPriority: '0'
```

Comment:
I think this just doesn't work and indeed I see that your "runtime" pipeline run didn't even start due to this. I concur this is a super-annoying limitation of the pipeline YAML but I'm afraid that we just need to live with it for now. Basically the only way you could make this work would be to create yet another helper pipeline template as a middle-man between the build-test-job and xplat-pipeline-job, synthesizing <code>testPriority</code> as an extra argument to the "middle template" that would subsequently construct the name and display name for <code>xplat-pipeline-job</code>. In this particular case I'm afraid it's easier to just bite the bullet and live with the exponential explosion, expanding the two conditional statements to four. To make matters more annoying, there's even no such thing as the C <code>?:</code> operator that would let you squash the testPriority variants and use it to translate their values as appropriate.

Reply:
This was PR working prior to the merge conflict.  When I rebased I was annoyed by the `exponentential explosion` so I tried to introduce `testPriority`.  As you note it is not working.  I will revise it.

Reply:
I apologize about the exponential explosion. At some point last fall, AzDO folks actually asked me to provide feedback on pipeline YAML pain points and I think this was the first thing I pointed out. The discussion was taking place on an internal e-mail thread I can forward to you if you're interested ;-).